### PR TITLE
Upgrade github.com/pingcap/tidb/pkg/parser

### PR DESCRIPTION
**Pull request by Tang8330**
It was using an old version of `tidb/pkg/parser` that was causing Linux failures when building a Go binary.

```
   • building                                       binary=dist/artie-reader_linux_386/reader
  ⨯ release failed after 2s                  error=failed to build for linux_386: exit status 1: # github.com/pingcap/tidb/pkg/parser/mysql
../../../../pkg/mod/github.com/pingcap/tidb/pkg/parser@v0.0.0-20231103042308-035ad5ccbe67/mysql/const.go:459:2: cannot use 1 << iota (untyped int constant 2147483648) as SQLMode value in constant declaration (overflows)
../../../../pkg/mod/github.com/pingcap/tidb/pkg/parser@v0.0.0-20231103042308-035ad5ccbe67/mysql/const.go:460:2: cannot use 1 << iota (untyped int constant 4294967296) as SQLMode value in constant declaration (overflows)
../../../../pkg/mod/github.com/pingcap/tidb/pkg/parser@v0.0.0-20231103042308-035ad5ccbe67/mysql/util.go:44:19: cannot use 4294967295 (untyped int constant) as int value in struct literal (overflows)
../../../../pkg/mod/github.com/pingcap/tidb/pkg/parser@v0.0.0-20231103042308-035ad5ccbe67/mysql/util.go:45:19: cannot use 4294967295 (untyped int constant) as int value in struct literal (overflows)

make: *** [release] Error 1
```

### CODE REVIEW

**Comment by dveeden** [comment]
Seems related to https://github.com/go-mysql-org/go-mysql/issues/929

**Comment by Tang8330** [comment]
Go version would be ideal to follow the release policy and that seems like there's no activity on that.

I'm particularly interested in getting the dependency updated so it can unbreak our builds.

@dveeden Any ideas on how we can get this in?

**Comment by dveeden** [comment]
I don't fully get why your build breaks and mine (and CI) doesn't?

**Comment by Tang8330** [comment]
>  I don't fully get why your build breaks and mine (and CI) doesn't?

Hm, I'm not sure how to answer this q. This broke on our repo: https://github.com/artie-labs/reader, running `make release` and the stacktrace is in the description of the PR.

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/go-mysql-org/go-mysql
 
-go 1.18
+go 1.22
```
Comment (dveeden) [inline go.mod:3]: So 1.22 and 1.23 are in supported status now and 1.21 is not anymore. 

https://go.dev/doc/devel/release

**Review by dveeden — APPROVED** [review]

**Comment by dveeden** [comment]
> > I don't fully get why your build breaks and mine (and CI) doesn't?
> 
> Hm, I'm not sure how to answer this q. This broke on our repo: https://github.com/artie-labs/reader, running `make release` and the stacktrace is in the description of the PR.

I tried a `make release` but that didn't work as I didn't have `goreleaser` installed. After that it complained about `error=missing GITHUB_TOKEN, GITLAB_TOKEN and GITEA_TOKEN` and then I tried `make build` which resulted in `rror=git tag v0.1.96 was not made against commit 04f3a34111bc8b9c750882329ad03b3a279b48e6`. Finally I tried `go build` which resulted in a `reader` binary:

```
$ go version -m reader | grep -E '^\s*build|tidb/pkg/parser'
	dep	github.com/pingcap/tidb/pkg/parser	v0.0.0-20231103042308-035ad5ccbe67	h1:m0RZ583HjzG3NweDi4xAcK54NBBPJh+zXp5Fp60dHtw=
	build	-buildmode=exe
	build	-compiler=gc
	build	CGO_ENABLED=1
	build	CGO_CFLAGS=
	build	CGO_CPPFLAGS=
	build	CGO_CXXFLAGS=
	build	CGO_LDFLAGS=
	build	GOARCH=amd64
	build	GOOS=linux
	build	GOAMD64=v3
	build	vcs=git
	build	vcs.revision=04f3a34111bc8b9c750882329ad03b3a279b48e6
	build	vcs.time=2024-11-17T17:42:29Z
	build	vcs.modified=false
```

**Comment by dveeden** [comment]
This fails for a 32-bit build.

```
$ GOARCH=386 go build
# github.com/pingcap/tidb/pkg/parser/mysql
/home/dvaneeden/go/pkg/mod/github.com/pingcap/tidb/pkg/parser@v0.0.0-20231103042308-035ad5ccbe67/mysql/const.go:459:2: cannot use 1 << iota (untyped int constant 2147483648) as SQLMode value in constant declaration (overflows)
/home/dvaneeden/go/pkg/mod/github.com/pingcap/tidb/pkg/parser@v0.0.0-20231103042308-035ad5ccbe67/mysql/const.go:460:2: cannot use 1 << iota (untyped int constant 4294967296) as SQLMode value in constant declaration (overflows)
/home/dvaneeden/go/pkg/mod/github.com/pingcap/tidb/pkg/parser@v0.0.0-20231103042308-035ad5ccbe67/mysql/util.go:44:19: cannot use 4294967295 (untyped int constant) as int value in struct literal (overflows)
/home/dvaneeden/go/pkg/mod/github.com/pingcap/tidb/pkg/parser@v0.0.0-20231103042308-035ad5ccbe67/mysql/util.go:45:19: cannot use 4294967295 (untyped int constant) as int value in struct literal (overflows)
```

I still think we should upgrade this package, but you might also want to check if you intended to build 32-bit builds instead of 64-bit

**Comment by dveeden** [comment]
@Tang8330 you might want to send a PR to add Artie Reader to this list: https://github.com/shlomi-noach/awesome-mysql

**Review by lance6716 — APPROVED** [review]

**Pull request merged**